quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java

```diff
@@ -0,0 +1,98 @@
+package com.upgrad.quora.api.controller;
+
+import com.upgrad.quora.api.model.*;
+import com.upgrad.quora.service.business.QuestionBusinessService;
+import com.upgrad.quora.service.entity.QuestionEntity;
+import com.upgrad.quora.service.exception.AuthorizationFailedException;
+import com.upgrad.quora.service.exception.InvalidQuestionException;
+import com.upgrad.quora.service.exception.UserNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/")
+public class QuestionController {
+
+    @Autowired
+    private QuestionBusinessService questionBusinessService;
+
+    // This method will create a new question posted by an authorized user.
+    @RequestMapping(method = RequestMethod.POST, path = "/question/create", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<QuestionResponse> createQuestion(
```

Comment:
i think you need to have user auth service to get the token and authenticate - 

This endpoint requests for all the attributes in 'QuestionRequest' about the question and access token of the signed in user as a string in the authorization field of the Request Header.

Reply:
Why is auth service required in this class? That's being taken care of the in the Question Business Service class.

Reply:
The controller will pass authorization parameter to the business service which will, in turn, validate the authorization before performing any operations.